### ISSUE TITLE
DynamicDependencies: Cleanup (Remove) obsolete DDLM packages

### ISSUE THREAD

**Issue by DrusTheAxe** [microsoft/ProjectReunion#592]
TL;DR Remove older redundant (obsolete) DynamicDependencies Lifetime Manager packages (aka DDLM packages aka PRddlm).

DDLM packages are named including their full version and architecture with a MinVersion dependency on the Project Reunion Framework package (aka PRfwk). When multiple compatible DDLM packages are present only the newest one is required.

For example, if Project Reunion version 1.0.0.0, 1.1.0.0 and 1.2.0.0 are installed for x64 we'll have three (3) PRddlm and three (3) PRfwk packages, but as PRfwk 1.2 is backwards compatible with PRfwk 1.1 and 1.0 Windows will eventually remove them leaving only PRfwk 1.2. Likewise, PRddlm 1.2 is backwards compatible with PRddlm 1.1 and 1.0 so the older packages are unnecessary (once not in use). However, PRddlm is uniquely named for each version+architecture so Windows sees distinct package family names with no relation, thus it thinks PRddlm-1.0.0.0-x64, PRddlm-1.1.0.0-x64 and PRddlm-1.2.0.0-x64 are the only packages in their family (which they are) and not remove any. Explicit action is required to remove any PRddlm.

PRddlm packages will collect over time, added to a system and never removed. In theory, yes, an app's uninstaller could explicitly remove a PRddlm, but an unpackaged app cannot know if any unpackaged app also needs the same PRddlm it's difficult for any one app to know it's safe to remove a PRddlm. To keep the installed PRddlm packages to a minimum we will provide a mechanism to periodically look for and remove redundant (unnecessary) PRddlm packages.

This may be handled via a BackgroundTask defined for the PRmain package that runs periodically (e.g. every 24 hours). This will call a PRddlm Garbage Collection (GC) API provided by Project Reunion.

See [5.4.3. Dynamic Dependency Lifetime Manager (DDLM)](https://github.com/microsoft/ProjectReunion/blob/main/specs/dynamicdependencies/DynamicDependencies.md#543-dynamic-dependency-lifetime-manager-ddlm) for more details about the DDLM packge.

**Comment by DrusTheAxe** [comment]
Completed
